bindings/pydeck/pydeck/bindings/deck.py

```diff
@@ -188,3 +188,8 @@ def to_html(
             **kwargs,
         )
         return f
+
+    def _repr_html_(self):
+        # doesn't actually need the HTML packaging in iframe_with_srcdoc,
+        # so we just take the HTML.data part
+        return self.to_html(notebook_display=True).data
```

Comment:
It's not clear to me why you need `.data`. Doesn't `.to_html()` work in general, so it should work inside `_repr_html_`?

Reply:
Its more like we don't need the `HTML` object that `to_html` returns. The `HTML` object is extra wrapping around the html text, which is all we really need.

I'm just using the minimal requirement here to make it easier for a future refactoring. See my comment above; we can avoid a lot of the code in this path, including the weird warning logic. I kept it out of this PR only to minimize the changes for now.